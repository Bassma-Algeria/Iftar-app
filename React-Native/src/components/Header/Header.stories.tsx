import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {Header} from './Header';

const headerStories = storiesOf('Header', module);

headerStories.add('h1', () => <Header variant="h1">إفطار</Header>);

headerStories.add('h2', () => <Header variant="h2">إفطار</Header>);

headerStories.add('h3', () => <Header variant="h3">إفطار</Header>);

headerStories.add('h4', () => <Header variant="h4">إفطار</Header>);

headerStories.add('h5', () => <Header variant="h5">إفطار</Header>);

headerStories.add('h6 - default', () => <Header>إفطار</Header>);
