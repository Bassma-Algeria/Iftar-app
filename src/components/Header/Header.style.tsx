import {StyleSheet} from 'react-native';

import {SPACING} from '../../theme/Spacings';

const FONTS = {
  regular: 'Cairo-Regular',
  medium: 'Cairo-Medium',
  semibold: 'Cairo-SemiBold',
  bold: 'Cairo-Bold',
  extraBold: 'Cairo-ExtraBold',
};

const styles = StyleSheet.create({
  h1: {
    fontFamily: FONTS.bold,
    fontSize: SPACING.larger,
  },

  h2: {
    fontFamily: FONTS.bold,
    fontSize: SPACING.large,
  },

  h3: {
    fontFamily: FONTS.semibold,
    fontSize: SPACING.xxl,
  },

  h4: {
    fontFamily: FONTS.regular,
    fontSize: SPACING.xl,
  },

  h5: {
    fontFamily: FONTS.regular,
    fontSize: 18,
  },

  h6: {
    fontFamily: FONTS.regular,
    fontSize: 16,
  },
});

export {styles, FONTS};
