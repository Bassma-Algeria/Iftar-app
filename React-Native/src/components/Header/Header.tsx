import React from 'react';
import {GestureResponderEvent, StyleProp, Text, TextStyle} from 'react-native';

import {FONTS, styles} from './Header.style';

import {COLORS} from '../../theme/Colors';

interface Props {
  variant?: keyof typeof styles;
  color?: keyof typeof COLORS;
  fontWeight?: keyof typeof FONTS;
  style?: StyleProp<TextStyle>;
  align?: 'auto' | 'left' | 'right' | 'center' | 'justify';
  onPress?: (event: GestureResponderEvent) => void;
}

const Header: React.FC<Props> = ({
  children,
  variant = 'h6',
  color = 'brown',
  fontWeight,
  onPress,
  align,
  style,
}) => {
  let elementStyles: StyleProp<TextStyle> = {
    ...styles[variant],
    color: COLORS[color],
  };

  if (fontWeight) {
    elementStyles.fontFamily = FONTS[fontWeight];
  }

  if (align) {
    elementStyles.textAlign = align;
  }

  return (
    <Text onPress={onPress} style={[elementStyles, style]}>
      {children}
    </Text>
  );
};

export {Header};
