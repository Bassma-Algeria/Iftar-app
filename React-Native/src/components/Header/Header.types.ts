import {GestureResponderEvent, StyleProp, TextStyle} from 'react-native';

import {FONTS, styles} from './Header.style';

import {COLORS} from '../../theme/Colors';

export interface Props {
  variant?: keyof typeof styles;
  color?: keyof typeof COLORS;
  fontWeight?: keyof typeof FONTS;
  style?: StyleProp<TextStyle>;
  onPress?: (event: GestureResponderEvent) => void;
}
