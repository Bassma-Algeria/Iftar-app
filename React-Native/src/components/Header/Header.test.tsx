import 'react-native';
import React from 'react';
import rendered from 'react-test-renderer';
import {Text} from 'react-native';
import {fireEvent, render} from '@testing-library/react-native';

import {Header} from './Header';

describe('Header Component', () => {
  it('should renders correctly with the default styles', () => {
    const tree = rendered.create(<Header>Hello</Header>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to change the color', () => {
    const tree = rendered.create(<Header color="black">Hello</Header>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have mutiple variant and change the size depend on the variant', () => {
    const tree = rendered.create(<Header variant="h2">Hello</Header>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to change the font weight', () => {
    const tree = rendered
      .create(<Header fontWeight="extraBold">Hello</Header>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to change add custom styles', () => {
    const tree = rendered
      .create(<Header style={{backgroundColor: '#000'}}>Hello</Header>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should run the function passed in the props when pressed', () => {
    const mockFunction = jest.fn();
    const instance = render(<Header onPress={mockFunction}>Click</Header>);

    fireEvent(instance.UNSAFE_getByType(Text), 'onPress');

    expect(mockFunction).toBeCalledTimes(1);
  });
});
