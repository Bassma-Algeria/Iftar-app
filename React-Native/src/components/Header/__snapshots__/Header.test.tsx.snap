// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should be able to change the color 1`] = `
<Text
  style={
    Object {
      "color": "#000",
      "fontFamily": "Cairo-Regular",
      "fontSize": 16,
    }
  }
>
  Hello
</Text>
`;

exports[`Header Component should be able to change the font weight 1`] = `
<Text
  style={
    Object {
      "color": "#7C4908",
      "fontFamily": "Cairo-ExtraBold",
      "fontSize": 16,
    }
  }
>
  Hello
</Text>
`;

exports[`Header Component should have mutiple variant and change the size depend on the variant 1`] = `
<Text
  style={
    Object {
      "color": "#7C4908",
      "fontFamily": "Cairo-Bold",
      "fontSize": 30,
    }
  }
>
  Hello
</Text>
`;

exports[`Header Component should renders correctly with the default styles 1`] = `
<Text
  style={
    Object {
      "color": "#7C4908",
      "fontFamily": "Cairo-Regular",
      "fontSize": 16,
    }
  }
>
  Hello
</Text>
`;
